includes/abstracts/llms-abstract-admin-wizard.php

```diff
@@ -0,0 +1,390 @@
+
+	/**
+	 * Get slug if prev step
+	 *
+	 * @since [version]
+	 *
+	 * @param string $step Step to use as current.
+	 * @return string
+	 */
+	public function get_prev_step( string $step = '' ): string {
+		$step = $step ?: $this->get_current_step();
+		$keys = array_keys( $this->get_steps() );
+		$i    = array_search( $step, $keys, true );
+
+		if ( false === $i || $i - 1 < 0 ) {
+			return false;
+		}
+
+		return $keys[ $i - 1 ] ?? '';
+	}
+
+	/**
+	 * Get the text to display on the "save" buttons
+	 *
+	 * @since [version]
+	 *
+	 * @param string $step Step to get text for.
+	 * @return string The translated text.
+	 */
+	private function get_save_text( string $step ): string {
+
+		/**
+		 * Filter the Save button text for a given step in the setup wizard
+		 *
+		 * The dynamic portion of this hook, `$step`, refers to the slug of the current step.
+		 *
+		 * @since [version]
+		 *
+		 * @param string $text Button text string.
+		 */
+		return apply_filters( "llms_setup_wizard_get_{$step}_save_text", $this->get_steps()[ $step ]['save'] ?? '' );
+	}
+
+	/**
+	 * Get the text to display on the "skip" buttons
+	 *
+	 * @since [version]
+	 *
+	 * @param string $step Step to get text for.
+	 * @return string Translated text.
+	 */
+	private function get_skip_text( string $step ): string {
+
+		/**
+		 * Filter the skip button text for a given step in the setup wizard
+		 *
+		 * The dynamic portion of this hook, `$step`, refers to the slug of the current step.
+		 *
+		 * @since [version]
+		 *
+		 * @param string $text Button text string.
+		 */
+		return apply_filters( "llms_setup_wizard_get_{$step}_skip_text", $this->get_steps()[ $step ]['skip'] ?? '' );
+
+	}
+
+	/**
+	 * Get the URL to a step
+	 *
+	 * @since [version]
+	 *
+	 * @param string $step Step slug.
+	 * @return string
+	 */
+	private function get_step_url( string $step ): string {
+		return add_query_arg(
+			array(
+				'page' => 'llms-' . $this->type,
+				'step' => $step,
+			),
+			admin_url()
+		);
+	}
+
+	/**
+	 * Get an array of step slugs => titles
+	 *
+	 * @since [version]
+	 *
+	 * @return array
+	 */
+	public function get_steps(): array {
+
+		/**
+		 * Filter the steps included in the setup wizard
+		 *
+		 * @since [version]
+		 *
+		 * @param string[] $steps Array of setup wizard steps. The array key is the slug/id of the step and the array value
+		 *                        is the step's title displayed in the wizard's navigation.
+		 */
+		return apply_filters( 'llms_setup_wizard_steps', $this->steps );
+
+	}
+
+	/**
+	 * Output the HTML content of the setup page
+	 *
+	 * @since [version]
+	 *
+	 * @return void
+	 */
+	public function output(): void {
+		$views_dir = trailingslashit( esc_attr( $this->views_dir ) );
+		$step_html = '';
+		$steps     = $this->get_steps();
+		$current   = $this->get_current_step() ?? 'intro';
+		$prev      = $this->get_prev_step();
+		$next      = $this->get_next_step();
+		$transient = get_transient( $this->transient_key ) ?? [];
```

Comment:
Not sure where we use this transient and what for?

Reply:
This was added for the cohorts wizard but might also be useful for other wizards, I think this is the only way to add it without overriding the output method